File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/AlarmStrategyPO.java
package io.iamcyw.ams.job.strategy.persistence;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

/**
 * alarm事件策略
 */
@Entity(name = "ALARM_STRATEGY")
public class AlarmStrategyPO extends PanacheEntity {

    public String name;

    @Column(name = "source")
    @ManyToOne(fetch = FetchType.LAZY)
    public AlarmSourcePO source;

    @Column(name = "alarm_policies")
    @ManyToMany(targetEntity = StrategyPolicyPO.class, fetch = FetchType.LAZY)
    @JoinTable(name = "ALARM_STRATEGY_POLICY_LINK",
            inverseJoinColumns = {@JoinColumn(name = "POLICY_ID", referencedColumnName = "ID")},
            joinColumns = {@JoinColumn(name = "STRATEGY_ID", referencedColumnName = "ID")})
    public List<StrategyPolicyPO> alarmPolicies;

    @Column(name = "level")
    @OneToMany(fetch = FetchType.LAZY)
    public Set<StrategyLevelPO> level;

    /**
     * 单位:分钟
     * <p>
     * 如果具体级别的间隔时间没有设置才使用该值
     */
    public Long levelTimeInterval = 0L;

    /**
     * 单位:秒
     * <p>
     * 默认警报可重复
     * <p>
     * 指定距离上次警报间隔多长才能再次发送
     */
    public Long repeatTimeInterval = 0L;

    @OneToMany(fetch = FetchType.LAZY)
    public Set<StrategyPushPO> push;

    @ManyToOne(fetch = FetchType.LAZY)
    public StrategySolvePO solve;


}

File: domain/src/jvmMain/kotlin/io/iamcyw/ams/domain/job/strategy/AlarmStrategyPO.kt
package io.iamcyw.ams.domain.job.strategy

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

/**
 * alarm事件策略
 */
@Entity(name = "ALARM_STRATEGY")
actual open class AlarmStrategyPO : PanacheEntityBase {
    companion object : PanacheCompanion<AlarmSourcePO>;

    @field:Id
    @field:GeneratedValue
    actual open var id: Long? = null

    @get:Column(name = "name")
    actual open var name: String = ""

    @get:Column(name = "source_rrn")
    actual open var source: AlarmSourcePO? = null

    @get:Column(name = "alarm_policies")
    actual open var alarmPolicies: MutableList<StrategyPolicyPO>? = null

    @get:Column(name = "level")
    actual open var level: MutableSet<StrategyLevelPO>? = null

}